### PR TITLE
Fix string datetime columns not appearing in `get_column_pair_plot`

### LINKED ISSUE #310: Method `get_column_pair_plot`: Does not plot synthetic data if datetime column is formatted as a string
### Environment Details

* SDMetrics version: 0.9.1
* Python version: Any
* Operating System: Any

### Error Description
Starting from SDV 1.0, users will provide a new metadata format. The current version of SDMetrics (0.9.1) already supports this format.

However, I am having an issue when the datetime columns are present in a string format such as: `"27 Dec 2020"` (format string =`"%d %b %Y"`)

### Steps to reproduce
Replicate using SDV 1.0

```python
from sdv.datasets.demo import download_demo
from sdv.lite import SingleTablePreset
from sdv.evaluation.single_table import get_column_pair_plot

real_data, metadata = download_demo(
    modality='single_table',
    dataset_name='fake_hotel_guests'
)

synthesizer = SingleTablePreset(metadata, name='FAST_ML')
synthesizer.fit(real_data)
synthetic_data = synthesizer.sample(num_rows=500)

fig = get_column_pair_plot(
    real_data=real_data,
    synthetic_data=synthetic_data,
    column_names=['checkin_date', 'amenities_fee'],
    metadata=metadata
)
```

Observe that only the real data shows up in the plot. The synthetic data is missing:

image

### PR DESCRIPTION
**Pull request by frances-h**
Resolve #310

### CODE REVIEW

**Review by amontanez24 — COMMENTED** [review]
Looks good! Just a few minor comments

File: sdmetrics/reports/utils.py [hunk 1]
```diff
@@ -433,9 +438,13 @@ def get_column_pair_plot(real_data, synthetic_data, column_names, metadata):
         return make_discrete_column_pair_plot(real_data, synthetic_data)
 
     if sdtypes[0] == 'datetime':
-        real_data.iloc[:, 0] = convert_to_datetime(real_data.iloc[:, 0])
+        datetime_format = col_meta[0].get('format') or col_meta[0].get('datetime_format')
+        real_data.iloc[:, 0] = convert_to_datetime(real_data.iloc[:, 0], datetime_format)
+        synthetic_data.iloc[:, 0] = convert_to_datetime(synthetic_data.iloc[:, 0], datetime_format)
```
[the remark this answers is not in the document]
Comment (frances-h) [reply sdmetrics/reports/utils.py]: It was a bit awkward to try to pull it out into a function so I switched it to a `for` loop instead. Let me know if you think a separate helper function would be better.

**Review by amontanez24 — APPROVED** [review]
Thanks for addressing!

**Review by pvk-developer — APPROVED** [review]
LGTM ! :shipit:

**Pull request merged**